cpp/daal/src/algorithms/pca/pca_dense_correlation_batch_impl.i

```diff
@@ -111,6 +114,7 @@ services::Status PCACorrelationKernel<batch, algorithmFPType, cpu>::compute(
             DAAL_CHECK_STATUS(status, this->copyVarianceFromCovarianceTable(covarianceTable, variances));
         }
 
+        if (needCorrelation)
         {
             DAAL_ITTNOTIFY_SCOPED_TASK(compute.full.correlationFromCovariance);
             DAAL_CHECK_STATUS(status, this->correlationFromCovarianceTable(covarianceTable));
```

Comment:
Can we compute this by setting:
```cpp
covarianceAlg->parameter.outputMatrixType = covariance::correlationMatrix;
```

Reply:
No, because we need to compute variances which can't be computed by correlation algorithm. So, firstly we compute covariance, secondly we compute variances from covariance and then correlation if needed.
Actually, we can compute means and variances from the original data and then compute covariance/correlation depending on `covarianceAlg->parameter.outputMatrixType`. But I don't think it's the best option from the performance perspective

Reply:
OK, but please restore prev value of outputMatrixType after computing of covariance.